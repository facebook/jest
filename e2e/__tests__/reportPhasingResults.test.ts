/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {skipSuiteOnJasmine} from '@jest/test-utils';
import {json} from '../runJest';

const MASKED_FOR_SNAPSHOT = 987654321;

skipSuiteOnJasmine();

describe('phasing result reporting', () => {
  it('has extra results', () => {
    const result = json('report-phasing-results', ['--reportPhasingResults']);
    expect(result.exitCode).toBe(0);
    expect(result.json.testResults).toHaveLength(1);
    const testResult = result.json.testResults[0];
    for (const assertionResult of testResult.assertionResults) {
      expect(assertionResult.duration).toBeGreaterThanOrEqual(0);
      assertionResult.duration = MASKED_FOR_SNAPSHOT;
    }
    expect(testResult.assertionResults).toMatchSnapshot();
  });
});
