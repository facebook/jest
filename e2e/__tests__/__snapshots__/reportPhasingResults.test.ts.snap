// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`phasing result reporting has extra results 1`] = `
Array [
  Object {
    "ancestorTitles": Array [
      "the test file",
    ],
    "duration": 987654321,
    "failureMessages": Array [],
    "fullName": "the test file parses, transpiles, and loads imports",
    "location": null,
    "status": "passed",
    "title": "parses, transpiles, and loads imports",
  },
  Object {
    "ancestorTitles": Array [
      "the test file",
    ],
    "duration": 987654321,
    "failureMessages": Array [],
    "fullName": "the test file produces tests",
    "location": null,
    "status": "passed",
    "title": "produces tests",
  },
  Object {
    "ancestorTitles": Array [
      "desc",
    ],
    "duration": 987654321,
    "failureMessages": Array [],
    "fullName": "desc beforeAll",
    "location": null,
    "status": "passed",
    "title": "beforeAll",
  },
  Object {
    "ancestorTitles": Array [
      "desc",
    ],
    "duration": 987654321,
    "failureMessages": Array [],
    "fullName": "desc beforeAll",
    "location": null,
    "status": "passed",
    "title": "beforeAll",
  },
  Object {
    "ancestorTitles": Array [
      "desc",
    ],
    "duration": 987654321,
    "failureMessages": Array [],
    "fullName": "desc does nothing",
    "location": null,
    "status": "passed",
    "title": "does nothing",
  },
  Object {
    "ancestorTitles": Array [
      "desc",
    ],
    "duration": 987654321,
    "failureMessages": Array [],
    "fullName": "desc does nothing again",
    "location": null,
    "status": "passed",
    "title": "does nothing again",
  },
  Object {
    "ancestorTitles": Array [
      "desc",
    ],
    "duration": 987654321,
    "failureMessages": Array [],
    "fullName": "desc afterAll",
    "location": null,
    "status": "passed",
    "title": "afterAll",
  },
]
`;
